Providers/Windows Accounts/scripts/account_discovery.ps1

```diff
@@ -1,216 +1,374 @@
+	}
+	
 }
 
-Try {
-    $Credential = New-Object System.Management.Automation.PSCredential @($LoginUsername, $LoginPassword)
-    $HostsArray = $Hosts -split "[ ,;]"
-    
-    Set-ExecutionPolicy -ExecutionPolicy Bypass -Scope Process
-    Import-Module NetTCPIP
-    
-    #If a single host is specified and that host is also listening on port 636 it is likely a domain controller, in which case enumerating local accounts would be redundant
-    #So instead enumerate active domain computers for the list of hosts to query for local accounts
-    If (($HostsArray.Count -eq 1) -and ((Test-NetConnection $HostsArray[0] -Port 636).TcpTestSucceeded)) {
-        $DomainFQDN = $HostsArray[0]
-        #ForEach ($level in ($DomainFQDN -Split ("\\.")))
-        #{
-        #    $DomainDN += ",DC=" + $level
-        #}
-        #$DomainDN = $DomainDN.TrimStart(",")
-        
-        Try {
-            $ADSI = New-Object System.DirectoryServices.DirectoryEntry("LDAP://$DomainFQDN`:636", $Credential.UserName, $Credential.GetNetworkCredential().Password) -ErrorAction Stop
-            [void]$ADSI.ToString()
-        } catch [System.Management.Automation.RuntimeException] {
-            Write-Error "Unable to connect to $DomainDN"
-        } catch {
-            Write-Error $error[0].Exception.ToString()
-        }
-        
-        If ($ADSI.distinguishedName -ne "") {
-            $Searcher = New-Object -TypeName System.DirectoryServices.DirectorySearcher($ADSI)
-            $Searcher.Filter = "(&(objectclass=computer)" #Find only computer objects
-            $Searcher.Filter += "(!useraccountcontrol:1.2.840.113556.1.4.804:=2)" #Exclude disable accounts
-            $Searcher.Filter += "(!userAccountControl:1.2.840.113556.1.4.803:=8192)" #Exclude domain controllers
-            $Searcher.Filter += "(!serviceprincipalname=*MSClusterVirtualServer*)" #Exclude MS Clustering objects
-            If ($HostsLDAPSearchFilter) {
-                $Searcher.Filter += $HostsLDAPSearchFilter #Append any additional search filter from provider definition
-            }
-            $Searcher.Filter += ")"
+function Get-WinRMNetworkParameters
+{
+	param (
+		[Parameter(Mandatory = $True)]
+		[ValidateNotNullOrEmpty()]
+		[String]$HostName)
+	
+	Import-Module NetTCPIP
+	switch ($HostName)
+	{
+		({ (Resolve-DnsName -Name $_ -ErrorAction SilentlyContinue).IPAddress })
+		{
+			$SessionParameters = @{
+				ComputerName = $Hostname
+			}
+			
+			switch ($_)
+			{
+				({
+						#check if host supports SSL Port
+						[System.Net.Sockets.TcpClient]::new().ConnectAsync($_, 5986).Wait(100)
+					})
+				{
+					$SessionParameters.Add("Port", 5986)
+					$SessionParameters.Add("UseSSL", $True)
+					Try
+					{
+						$result = Test-WSMan @SessionParameters
+					}
+					Catch
+					{
+						Write-EventLog -LogName Devolutions -EntryType Warning -Source "DVLS" -EventId 1 -Message $Error[0].Exception.ToString()
+						#Write-Error $Error[0].Exception.ToString()
```

Comment:
Referencing the DVLS event source will fail if it doesn't already exist. Since it wasn't in the original, I'd just remove these references and replaced with Write-Verbose/Warning/Error.